default_settings.py

```diff
@@ -156,5 +156,9 @@
         if "excluded" in streams and streams["excluded"]:
             excluded_streams = streams["excluded"]
 
+        include_web_public_streams = True
```

Comment:
Tested: https://rht.github.io/zulip-archive-sample/. I only explicitly specified "helpdesk (published)" as the included stream. And when I set this line to False, only this stream was archived. When I set this line to True, now there are additionally "announcements", "sciml-bridged", and "diffeq-bridged" in the archive. Tho this means that I haven't end-to-end tested configuring from streams.yaml instead of here.

Reply:
Currently waiting the test result of setting `include_web_public_streams` to be False in streams.yaml.

Reply:
It gave the wrong behavior, I just discovered that `include_web_public_streams: False` in YAML file is parsed as `{'include_web_public_streams': 'False'}`. There are various options to convert the 'False' to boolean in https://stackoverflow.com/questions/36463531/pyyaml-automatically-converting-certain-keys-to-boolean-values. But this is just not baked in the library. Should I just move it to [settings.py](https://github.com/zulip/zulip-archive/blob/master/default_settings.py) instead of [streams.yaml](https://github.com/zulip/zulip-archive/blob/master/default_streams.yaml), therefore constraining it to `True`/`False`?

Reply:
Also, the default behavior should probably to not archive web-public streams by default? Otherwise, this breaks compatibility, and users will observe new streams being archived when they update their GH Actions.

Reply:
Yeah; I think we may want to change the default down the line.

Reply:
Idunno, maybe it makes more sense to just set `include_web_public_streams` as defaulting to true, and do a major version number bump and bit of documentation around that change?

I feel like that probably will be what most new projects using this want.

Reply:
I don't really understand the split between `settings.py` and the YAML configuration file; can you explain that spect of the current setup?

Reply:
> I think it's possible if you put true in the YAML file, rather than "True", it'll import properly as a boolean.

I just tried `true`. It is still parsed as a string.

Regarding with the split, it was @hackerkid who did it in https://github.com/zulip/zulip-archive/commit/bd2fc217c0479395d6046a961575ec7b7f57918c. I'm not sure of the rationale.